### PR TITLE
Switch to BinaryBuilder + drop support for 0.7

### LINKED ISSUE #44: Drop 0.6, 0.7?
0.6 and 0.7 are now officially unmaintained, https://discourse.julialang.org/t/julia-0-6-and-0-7-are-now-officially-un-maintained/20449. Maybe a good time to stop testing on 0.6 and 0.7 and require 1.0?

### LINKED ISSUE #51: Switch from BinDeps to BinaryBuilder
BinDeps is deprecated. Example of how to use [BinaryBuilder](https://github.com/JuliaPackaging/BinaryBuilder.jl) in [CBC](https://github.com/JuliaOpt/Cbc.jl/blob/master/deps/build.jl).

Related to #45. We should use shared libraries in all the platforms from now on.

### PR DESCRIPTION
**Pull request by bstellato**
Fixes #44 
Fixes #51 

Replaces #49

### CODE REVIEW

**Comment by bstellato** [comment]
I am getting the error
```
Got exception outside of a @test MathOptInterface.UnsupportedAttribute{MathOptInterface.VariablePrimalStart}: Attribute MathOptInterface.VariablePrimalStart() is not supported by the model.
```

@blegat do you know why it happens? It looks like something changed in MOI. I suspect it happened in MOI `v0.8.2` wher eyou added a test in `contlineartest` to partially set `VariablePrimalStart`.

**Comment by bstellato** [comment]
Ok I have removed it for now. It looks like it is not required for OSQP from this comment

https://github.com/JuliaOpt/MathOptInterface.jl/blob/ecf691545e67552ff437ed26ec4ddfff03c50327/src/Test/contlinear.jl#L1715-L1716

File: REQUIRE [hunk 1]
```diff
@@ -1,5 +1,5 @@
-BinDeps
-julia 0.6
+julia 1.0
```
Comment (mlubin) [inline REQUIRE]: In order to tag a new release on Julia 1.0, you have to move to the Project.toml format, which replaces REQUIRE: https://discourse.julialang.org/t/switching-package-registration-systems-soon/22677

**Comment by bstellato** [comment]
Thanks! I have made the changes you suggested.  I kept Compat in the test dependencies because it is needed in MathProgBase.

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name = "OSQP"
+uuid = "ab2f91bb-94b4-55e3-9ba0-7f65df51de79"
+keywords = ["numerical-optimization", "solver"]
+license = "Apache 2.0"
+desc = "The Operator Splitting QP Solver."
+version = "0.5.1"
```
Comment (blegat) [inline Project.toml]: You can add `repo = "https://github.com/oxfordcontrol/OSQP.jl.git"`

**Pull request merged**